src/codec_aom.c

```diff
@@ -512,12 +512,18 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
 #endif
         int aomCpuUsed = -1;
         if (encoder->speed != AVIF_SPEED_DEFAULT) {
-            if (encoder->speed < 8) {
-                aomCpuUsed = AVIF_CLAMP(encoder->speed, 0, 6);
-            } else {
+            aomCpuUsed = AVIF_CLAMP(encoder->speed, 0, 9);
+#if defined(AOM_USAGE_ALL_INTRA) && \
+    AOM_ENCODER_ABI_VERSION >= (9 + AOM_CODEC_ABI_VERSION + AOM_EXT_PART_ABI_VERSION)
+            // Speeds 7-9 in all intra mode work.
+            if (!(addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE) && (aomCpuUsed >= 7)) {
+                aomUsage = AOM_USAGE_REALTIME;
+            }
+#else
+            if (aomCpuUsed >= 7) {
                 aomUsage = AOM_USAGE_REALTIME;
-                aomCpuUsed = AVIF_CLAMP(encoder->speed - 2, 6, 8);
             }
+#endif
```

Comment:
Ignoring comments, we can restructure lines 515-526 as follows:

```
            aomCpuUsed = AVIF_CLAMP(encoder->speed, 0, 9);
            if (aomCpuUsed >= 7) {
#if defined(AOM_USAGE_ALL_INTRA) && AOM_ENCODER_ABI_VERSION >= (10 + AOM_CODEC_ABI_VERSION + AOM_EXT_PART_ABI_VERSION)
                if (!(addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE)) {
                    aomUsage = AOM_USAGE_REALTIME;
                }
#else
                aomUsage = AOM_USAGE_REALTIME;
#endif
            }
```

**NOTE:** Please run

`clang-format -i --style=file src/codec_aom.c`

Reply:
Done